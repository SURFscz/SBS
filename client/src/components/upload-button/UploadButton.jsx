import React from "react";
import "../input-field/InputField.scss";
import Button from "../button/Button";

export default function UploadButton({
                                         name,
                                         onFileUpload = null,
                                         txt,
                                         acceptFileFormat = "text/csv"
                                     }) {
    let fileInput;

    const onClick = () => {
        fileInput.click();
    };

    return (
        <div className="file-upload-button-container">
            <input type="file"
                   id={`fileUpload_${name}`}
                   ref={ref => fileInput = ref}
                   name={`fileUpload_${name}`}
                   accept={acceptFileFormat}
                   style={{display: "none"}}
                   onChange={onFileUpload}/>
            <Button txt={txt} onClick={onClick}/>
        </div>
    );

}
