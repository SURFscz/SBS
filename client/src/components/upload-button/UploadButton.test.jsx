import React from "react";
import {render} from "@testing-library/react";
import UploadButton from "./UploadButton";

describe("UploadButton", () => {
    it("matches snapshot with default props", () => {
        const {asFragment} = render(
            <UploadButton
                onChange={() => {}}
                accept="image/*"
                disabled={false}
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it("matches snapshot with custom label", () => {
        const {asFragment} = render(
            <UploadButton
                onChange={() => {}}
                accept="image/*"
                disabled={false}
                label="Custom Upload"
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it("matches snapshot when disabled", () => {
        const {asFragment} = render(
            <UploadButton
                onChange={() => {}}
                accept="image/*"
                disabled={true}
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
