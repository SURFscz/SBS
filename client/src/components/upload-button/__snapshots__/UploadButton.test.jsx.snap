// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadButton matches snapshot when disabled 1`] = `
<DocumentFragment>
  <div
    class="file-upload-button-container"
  >
    <input
      accept="text/csv"
      id="fileUpload_undefined"
      name="fileUpload_undefined"
      style="display: none;"
      type="file"
    />
    <button
      class="sds--btn sds--btn--primary     "
      type="button"
    />
  </div>
</DocumentFragment>
`;

exports[`UploadButton matches snapshot with custom label 1`] = `
<DocumentFragment>
  <div
    class="file-upload-button-container"
  >
    <input
      accept="text/csv"
      id="fileUpload_undefined"
      name="fileUpload_undefined"
      style="display: none;"
      type="file"
    />
    <button
      class="sds--btn sds--btn--primary     "
      type="button"
    />
  </div>
</DocumentFragment>
`;

exports[`UploadButton matches snapshot with default props 1`] = `
<DocumentFragment>
  <div
    class="file-upload-button-container"
  >
    <input
      accept="text/csv"
      id="fileUpload_undefined"
      name="fileUpload_undefined"
      style="display: none;"
      type="file"
    />
    <button
      class="sds--btn sds--btn--primary     "
      type="button"
    />
  </div>
</DocumentFragment>
`;
